core/domain/suggestion_jobs_one_off.py

```diff
@@ -369,3 +370,54 @@ def _update_community_contribution_stats_transactional(
 
         # Only yield the values from the transactions.
         yield (key_from_transaction, count_value_from_transaction)
+
+
+class PopulateFinalReviewerIdOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to populate final_reviewer_id property in the
+    suggestion models which does not have but expected to have one.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        if not (item.status in [
+                suggestion_models.STATUS_ACCEPTED,
+                suggestion_models.STATUS_REJECTED] and (
+                    item.final_reviewer_id is None)):
+            return
+
+        message_status = feedback_models.STATUS_CHOICES_FIXED
+        if item.status == suggestion_models.STATUS_REJECTED:
+            message_status = feedback_models.STATUS_CHOICES_IGNORED
+
+        message_model_class = feedback_models.GeneralFeedbackMessageModel
+        message_models = message_model_class.query(
+            message_model_class.thread_id == item.id,
+            message_model_class.updated_status == message_status).fetch()
+
+        if not message_models:
+            yield ('FAILED_NONE_MESSAGE_MODEL', (item.id, item.status))
+            return
+
+        if len(message_models) != 1:
+            yield ('FAILED_MULTIPLE_MESSAGE_MODEL', (item.id, item.status))
```

Comment:
Actually this might be possible. you can edit the status multiple times. What you should consider is the last model which updates the status.